packages/editor/src/browser/editor-decoration.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { Range } from "./editor";
+import { EditorManager } from "./editor-manager";
+import URI from "@theia/core/lib/common/uri";
+
+export const EditorDecorationsService = Symbol('EditorDecorationsService');
+
+export interface EditorDecorationsService {
+    registerDecorationType(decorationType: DecorationType): void;
+    removeDecorationType(key: string): void;
+    getDecorationTypes(): DecorationType[];
```

Comment:
Do we need an event if decoration types changed?

Reply:
actually, I'm not sure, if `removeDecorationType` is necessary at all. is there a use case, where we want to have this dynamic behavior? maybe it should be just a contribution of `DecorationType`s?

Reply:
I am fine with getting rid of `removeDecorationType` but we should be clear about when new decoration types should be registered, i.e. if you register after the editor is opened then they won't be taken into the account.

Doing it in the constructor or in `postConstruct` hook of the frontend application contribution seems to be the only safe place.

Reply:
thanks for that hint!